src/scripts/main.js

```diff
@@ -1 +1,79 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
```

Comment:
The timeout for rejecting the first promise is set to 3 seconds. Consider whether this duration is appropriate for your application's user experience. If users might take longer to click, you might want to increase this duration.